src/astUtils/visitors.ts

```diff
@@ -33,137 +78,51 @@ export function createStatementVisitor(
+        ClassFieldStatement?: (statement: ClassFieldStatement, parent: Statement) => Statement | void;
+        //expressions
+        BinaryExpression?: (expression: BinaryExpression, parent: Statement | Expression) => Expression | void;
+        CallExpression?: (expression: CallExpression, parent: Statement | Expression) => Expression | void;
+        FunctionExpression?: (expression: FunctionExpression, parent: Statement | Expression) => Expression | void;
+        NamespacedVariableNameExpression?: (expression: NamespacedVariableNameExpression, parent: Statement | Expression) => Expression | void;
+        DottedGetExpression?: (expression: DottedGetExpression, parent: Statement | Expression) => Expression | void;
+        XmlAttributeGetExpression?: (expression: XmlAttributeGetExpression, parent: Statement | Expression) => Expression | void;
+        IndexedGetExpression?: (expression: IndexedGetExpression, parent: Statement | Expression) => Expression | void;
+        GroupingExpression?: (expression: GroupingExpression, parent: Statement | Expression) => Expression | void;
+        LiteralExpression?: (expression: LiteralExpression, parent: Statement | Expression) => Expression | void;
+        EscapedCharCodeLiteralExpression?: (expression: EscapedCharCodeLiteralExpression, parent: Statement | Expression) => Expression | void;
+        ArrayLiteralExpression?: (expression: ArrayLiteralExpression, parent: Statement | Expression) => Expression | void;
+        AALiteralExpression?: (expression: AALiteralExpression, parent: Statement | Expression) => Expression | void;
+        UnaryExpression?: (expression: UnaryExpression, parent: Statement | Expression) => Expression | void;
+        VariableExpression?: (expression: VariableExpression, parent: Statement | Expression) => Expression | void;
+        SourceLiteralExpression?: (expression: SourceLiteralExpression, parent: Statement | Expression) => Expression | void;
+        NewExpression?: (expression: NewExpression, parent: Statement | Expression) => Expression | void;
+        CallfuncExpression?: (expression: CallfuncExpression, parent: Statement | Expression) => Expression | void;
+        TemplateStringQuasiExpression?: (expression: TemplateStringQuasiExpression, parent: Statement | Expression) => Expression | void;
+        TemplateStringExpression?: (expression: TemplateStringExpression, parent: Statement | Expression) => Expression | void;
+        TaggedTemplateStringExpression?: (expression: TaggedTemplateStringExpression, parent: Statement | Expression) => Expression | void;
     }
 ) {
-    return (statement: Statement, parent: Statement): Statement | void => {
+    return <WalkVisitor>((statement: Statement, parent: Statement): Statement | void => {
         return visitor[statement.constructor.name]?.(statement, parent);
-    };
-}
-
-/**
- * Create a statement -> expression visitor
- */
-export function createStatementExpressionsVisitor(
```

Comment:
This functionality has been moved into the AST nodes (`Expression` and `Statement`). 

**TODO: ** Add `WalkOptions` flag for _only_ walking own expressions (right now it'll still walk sub statements and expressions).